air/src/air/coefficients.rs

```diff
@@ -73,16 +73,38 @@ pub struct LagrangeConstraintsCompositionCoefficients<E: FieldElement> {
 ///
 /// 1. The evaluation proofs for each trace polynomial at $z$ and $g \cdot z$ can be batched using
 /// the non-normalized Lagrange kernel over the set $\{z, g \cdot z\}$. This, however, requires
-/// that the FRI protocol is run with rate $\rho^{+} := \frac{\kappa + 2}{\nu}$ where $\kappa$ and
-/// $\nu$ are the length of the execution trace and the LDE domain size, respectively.
+/// that the FRI protocol is run with a larger agreement parameter
+/// $\alpha^{+} = (1 + 1/2m)\cdot\sqrt{\rho^{+}}$ where $\rho^{+} := \frac{\kappa + 2}{\nu}$,
+/// $\kappa$ and $\nu$ are the length of the execution trace and the LDE domain size, respectively.
 /// 2. The resulting $Y(x)$ do not need to be degree adjusted but the soundness error of the
 /// protocol needs to be updated. For most combinations of batching parameters, this leads to a
 /// negligible increase in soundness error. The formula for the updated error can be found in
 /// Theorem 8 of https://eprint.iacr.org/2022/1216.
+///
+/// In the case when the trace polynomials contain a trace polynomial corresponding to a Lagrange
+/// kernel column, the above expression of $Y(x)$ includes the additional term given by
+///
+/// $$
+/// \gamma \cdot \frac{T_l(x) - p_S(x)}{Z_S(x)}
+/// $$
+///
+/// where:
+///
+/// 1. $\gamma$ is the composition coefficient for the Lagrange kernel trace polynomial.
+/// 2. $T_l(x) is the evaluation of the Lagrange trace polynomial at $x$.
+/// 3. $S$ is the set of opening points for the Lagrange kernel i.e.,
+/// $S := {z, z.g, z.g^2, ..., z.g^{2^{log_2(\nu) - 1}}}$.
+/// 4. $p_S(X)$ is the polynomial of minimal degree interpolating the set ${(a, T_l(a)): a \in S}$.
+/// 5. $Z_S(X)$ is the polynomial of minimal degree vanishing over the set $S$.
+///
+/// Note that, if a Lagrange kernel trace polynomial is present, then $\rho^{+}$ from above should
+/// be updated to be $\rho^{+} := \frac{\kappa + log_2(\nu) + 1}{\nu}$.
 #[derive(Debug, Clone)]
 pub struct DeepCompositionCoefficients<E: FieldElement> {
     /// Trace polynomial composition coefficients $\alpha_i$.
     pub trace: Vec<E>,
     /// Constraint column polynomial composition coefficients $\beta_j$.
     pub constraints: Vec<E>,
+    /// Lagrange kernel trace polynomial composition coefficient $\gamma$.
+    pub lagrange_cc: Option<E>,
```

Comment:
nit: I would probably rename this to `lagrange_kernel`.

Reply:
Done